OutboundCallReminder/src/main/java/com/communication/outboundcallreminder/Ngrok/NgrokService.java

```diff
@@ -0,0 +1,107 @@
+package com.communication.outboundcallreminder.Ngrok;
+
+import org.json.simple.JSONArray;
+import java.io.*;
+import java.util.*;
+import com.communication.outboundcallreminder.Logger;
+
+public class NgrokService {
+    /// The NGROK process
+    private Process ngrokProcess;
+    // NgrokConnector connector;
+    private NgrokConnector connector = null;
+
+    public NgrokService(String ngrokPath, String authToken) {
+        connector = new NgrokConnector();
+        this.EnsureNgrokNotRunning();
+        this.CreateNgrokProcess(ngrokPath, authToken);
+    }
+
+    /// <summary>
+    /// Ensures that NGROK is not running.
+    /// </summary>
+    private void EnsureNgrokNotRunning() {
+        BufferedReader input = null;
+        String ngrokProcess = "ngrok.exe";
+        String processFilter = "/nh /fi \"Imagename eq " + ngrokProcess + "\"";
+        String tasksCmd = System.getenv("windir") + "/system32/tasklist.exe " + processFilter;
+        try {
+            Process process = Runtime.getRuntime().exec(tasksCmd);
+            input = new BufferedReader(new InputStreamReader(process.getInputStream()));
+
+            String line = null;
+            while ((line = input.readLine()) != null) {
+                // It means ngrok.exe running
+                if (line.contains(ngrokProcess)) {
+                    Logger.LogMessage(Logger.MessageType.INFORMATION, "Looks like NGROK is still running. Please kill it before running the provider again.");
+                    System.exit(0);
+                }
+            }
+            process.destroy();
+            input.close();
+        } catch (Exception ex) {
+            Logger.LogMessage(Logger.MessageType.ERROR ,"Not able to find the process -- > " + ex.getMessage());
+        }
+    }
+
+    /// <summary>
+    /// Kill ngrok.exe process
+    /// </summary>
+    public void Dispose() {
+        if (this.ngrokProcess != null) {
+            this.ngrokProcess.destroy();
+        }
+    }
+
+    /// <summary>
+    /// Creates the NGROK process.
+    /// </summary>
+    private void CreateNgrokProcess(String ngrokPath, String authToken) {
+        try {
+            String authTokenArgs = "";
+            if (authToken != null && !authToken.isEmpty()) {
+                authTokenArgs = "--authtoken " + authToken;
+            }
+
+            String openCmd = "cmd /c start cmd.exe /k ";
+            String fileName = ngrokPath + "ngrok.exe";
+            String arguments = " http http://localhost:9007/ -host-header=/localhost:9007/" + authTokenArgs;
```

Comment:
Don't we need a space before the 'authTokenArgs'. For example:
String arguments = " http http://localhost:9007/ -host-header=/localhost:9007/" + " " +authTokenArgs;

Reply:
Yes, correct, we need to add space. Since we are not passing authtoken, so authTokenArgs value is empty, so not causing any issue at the moment. Will add space before.